src/app/shared/components/template/services/instance/template-action.service.ts

```diff
@@ -195,13 +199,15 @@ export class TemplateActionService extends SyncServiceBase {
         // because a normal url starts with https://, the ':' separates it into a key and a value and the value
         // is sufficient for the url to launch.
         console.log("[GO TO URL]", { key, value });
-        // if there is no http or https then there is no : to separate and we only have a key in the url. This
-        // case then adds '//' to the key so it recognises it as external and not local
-        // This removes the need to have http or https in the url.
-        if (!value) {
-          value = "//" + key;
-        }
-        return this.templateNavService.handleNavActionExternal(value);
+        // form the url to generate the url from the key and value
+        // it does so by using a ternary operator to check if the key and value are present
+        // if so. it concatenates the key and value with a colon in between
+        // otherwise, it uses the key or value as the url (whichever is present will used, if both, then the key is taken)
+        // because sometimes when generating {key, value}, they may assign the url in just the key or value
```

Comment:
I think all the lines of comments from 199 to 206 could be re-worded to be more concise and clear. It could also be emphasised that this is a bit of a hack-y workaround by preceding it with the word "HACK:" Something like:

```ts
// HACK: URL strings that include the ":" character (as in `https://`) will be split on that character into a key-value pair. 
// If both a key and a value are populated, assume this to be the case and join on the ":" character.
```